### PR TITLE
 Added an `andMap`, similar to `Task.andMap`

### PR DESCRIPTION
**Pull request by hughfdjackson**
I really enjoy using `Task.andMap` and `Task.map` to gather together tasks into a function, and was a little disappointed you couldn't do it for a Signal.  

Do you think this is a decent addition to the library? :)

### CODE REVIEW

**Comment by jvoigtlaender** [comment]
Have you seen the `~` operator in `core`? 

http://package.elm-lang.org/packages/elm-lang/core/2.1.0/Signal#~

**Comment by hughfdjackson** [comment]
@jvoigtlaender That's a great point.  I've been mostly avoiding using infix operators (except `>>`, `<<`, `|>` and `<|`, which seem like the minimal set). 

I see that evan and team are of the same mind from their design guidelines (http://package.elm-lang.org/help/design-guidelines).  I think I might see if they'll accept a PR for this in core - and close this for the meantime.

**Comment by hughfdjackson** [comment]
@jvoigtlaender Hrm - reopening this, because the CONTRIBUTING.md in the core library says they prefer functionality to come via an extras library: https://github.com/elm-lang/core/blob/master/CONTRIBUTING.md#adding-new-functions. 

I still think it's valuable to have a commonly-named alternative to an infix operator for readability/searchability purposes.

File: src/Signal/Extra.elm [hunk 1]
```diff
@@ -34,6 +35,9 @@ module Signal.Extra
 # Flipped fancy map
 @docs (~>)
 
+# Mapping
+@docs andMap
+
```
Comment (Apanatshka) [inline src/Signal/Extra.elm:37]: Can you add the `(~>)` to this section and remove the **Flipped fancy map** section?

File: src/Signal/Extra.elm [hunk 2]
```diff
@@ -66,8 +70,30 @@ two!
 
 infixl 4 ~>
 
+{-| Apply a Signal of functions to another signal. Like `Task.andMap`, this
+provides a more attractive way to combine several signals together into a
```
Comment (Apanatshka) [inline src/Signal/Extra.elm]: Let's not put in a value judgement like "attractive", when you can objectively say that this is more scalable<sup>†</sup> than `mapN`. 
<sup>†</sup> or more easily extensible, smaller diffs, something like that

**Comment by Apanatshka** [comment]
Thanks for providing this. It's a good idea to add it :)
I've added some inline comments. 
Can you also write 1 or 2 tests to capture that the result signal should update whenever either of the two argument signals updates?

**Comment by hughfdjackson** [comment]
@Apanatshka :) Great - I think I've addressed all of your concerns.  Let me know what you think.

**Comment by Apanatshka** [comment]
You have :)
Thanks for this addition :+1:

**Pull request merged**